src/realm/group_shared.hpp

```diff
@@ -665,6 +681,30 @@ inline void SharedGroup::get_stats(size_t& free_space, size_t& used_space) {
     used_space = m_used_space;
 }
 
+template <typename TFunc>
+bool SharedGroup::delete_realm(const std::string path, TFunc cleanup) {
+    // auto coordination_dir = path + ".management";
+    auto lockfile_path(path + ".lock");
+    auto management_dir = path + ".management";
+
+    File lockfile;
+    lockfile.open(lockfile_path, File::access_ReadWrite, File::create_Auto, 0); // Throws
+    File::CloseGuard fcg(lockfile);
+
+    if (lockfile.try_lock_exclusive()) { // Throws
+        File::UnlockGuard ulg(lockfile);
+        std::remove(path.c_str()); // ignore result, file may not exist
```

Comment:
@finnschiermer realm-java has an API allows user to copy a prebuilt realm file first if the realm file doesn't exists. I am thinking maybe the remove can be called by binding, so this API can handle 2 cases:
1. delete realm. Binding just holds the lock and delete the realm file
2. copying when initialization. Binding just holds the lock, check if the file exists, copy the file if it doesn't. What do you think?

Reply:
Seems like you can just do 2 anyway if you want?

Reply:
But if `std::remove(path.c_str());` is called before the binding callback, I cannot do 2 anymore since the file will deleted anymore before binding check the existence?

Reply:
Seems like we should just do the callback, providing the path to the realm file and the path to the management directory, and leave it to the callback to clean up? I'll change the code.